python/sparkdl/param/converters.py

```diff
@@ -0,0 +1,155 @@
+            for _maybe_col_name, _maybe_tnsr_name in value.items():
+                # Check if the non-tensor value is of string type
+                _col_name = _get_strict_col_name(_maybe_col_name)
+                # Check if the tensor name is actually valid
+                _tnsr_name = _get_strict_tensor_name(_maybe_tnsr_name)
+                strs_pair_seq.append((_col_name, _tnsr_name))
+
+            return sorted(strs_pair_seq)
+
+        err_msg = "Could not convert [type {}] {} to column name to tf.Tensor name mapping"
+        raise TypeError(err_msg.format(type(value), value))
+
+    @staticmethod
+    def asTensorNameToColumnMap(value):
+        """
+        Convert a value to a :py:obj:`tf.Tensor` name to column name mapping
+        as a sorted list of string pairs, if possible.
+        """
+        if isinstance(value, dict):
+            strs_pair_seq = []
+            for _maybe_tnsr_name, _maybe_col_name in value.items():
+                # Check if the non-tensor value is of string type
+                _col_name = _get_strict_col_name(_maybe_col_name)
+                # Check if the tensor name is actually valid
+                _tnsr_name = _get_strict_tensor_name(_maybe_tnsr_name)
+                strs_pair_seq.append((_tnsr_name, _col_name))
+
+            return sorted(strs_pair_seq)
+
+        err_msg = "Could not convert [type {}] {} to tf.Tensor name to column name mapping"
+        raise TypeError(err_msg.format(type(value), value))
+
+    @staticmethod
+    def toTFHParams(value):
+        """ Convert a value to a :py:obj:`tf.contrib.training.HParams` object, if possible. """
+        if isinstance(value, tf.contrib.training.HParams):
+            return value
+        else:
+            raise TypeError("Could not convert %s to TensorFlow HParams" % type(value))
+
+    @staticmethod
+    def toStringOrTFTensor(value):
+        """ Convert a value to a str or a :py:obj:`tf.Tensor` object, if possible. """
+        if isinstance(value, tf.Tensor):
+            return value
+        try:
+            return TypeConverters.toString(value)
+        except Exception as exc:
+            err_msg = "Could not convert [type {}] {} to tf.Tensor or str. {}"
+            raise TypeError(err_msg.format(type(value), value, exc))
+
+    @staticmethod
+    def supportedNameConverter(supportedList):
+        """
+        Create a converter that try to check if a value is part of the supported list.
+
+        :param supportedList: list, containing supported objects.
+        :return: a converter that try to convert a value if it is part of the `supportedList`.
+        """
+        def converter(value):
+            if value in supportedList:
+                return value
+            err_msg = "[type {}] {} is not in the supported list: {}"
+            raise TypeError(err_msg.format(type(value), str(value), supportedList))
+
+        return converter
+
+    @staticmethod
+    def toKerasLoss(value):
```

Comment:
I agree with @thunterdb  on that function naming should be precise. However, the entire Spark MLlib Param class promotes this idea of a "TypeConverter", so I think it's actually more confusing if we start naming things not similarly to the ones in Spark MLlib (i.e., XXConverter, toYY)... Unless @jkbradley says otherwise :-P

Reply:
Talked with @jkbradley and he didn't have strong opinions, but since all of these functions return the input value if it is of type Y in toY(), we should keep the function names toY().

Reply:
ok